spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/DelegatingResourceLoader.java

```diff
@@ -137,12 +137,10 @@ public Resource getResource(String location) {
 				return new FileSystemResource(cachedResource);
 			}
 		}
-		catch (URISyntaxException e) {
```

Comment:
IMO that should produce an `IllegalArgumentException` that wraps the URI syntax Exception